gobblin-utility/src/main/java/gobblin/util/PropertiesUtils.java

```diff
@@ -48,4 +51,23 @@ public static Properties combineProperties(Properties... properties) {
   public static boolean getPropAsBoolean(Properties properties, String key, String defaultValue) {
     return Boolean.valueOf(properties.getProperty(key, defaultValue));
   }
+
+  /**
+   * Extract all the keys that start with a <code>prefix</code> in {@link Properties} to a new {@link Properties}
+   * instance.
+   *
+   * @param properties the given {@link Properties} instance
+   * @param prefix of keys to be extracted
+   * @return a {@link Properties} instance
+   */
+  public static Properties extractPropertiesWithPrefix(Properties properties, Optional<String> prefix) {
```

Comment:
Added some basic tests, there seems to be none pre-existing tests for most of the basic util methods. Will create a ticket for it as well.